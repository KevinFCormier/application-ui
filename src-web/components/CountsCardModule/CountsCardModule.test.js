/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

const React = require('react');
const renderer = require('react-test-renderer');
const CountsCardModule = require('../CountsCardModule')
  .default;

describe('CountsCardModule', () => {
  const countsCardData = [
    {
      msgKey: 'table.header.deployables',
      count: 3,
    },
    {
      msgKey: 'table.header.deployments',
      count: 1,
    },
    {
      msgKey: 'table.header.failedDeployments',
      count: 0,
    },
  ];

  it('CountsCardModule renders correctly.', () => {
    const tree = renderer.create(<CountsCardModule data={countsCardData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
