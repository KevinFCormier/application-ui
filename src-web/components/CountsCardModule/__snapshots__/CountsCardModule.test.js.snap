// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountsCardModule CountsCardModule renders correctly. 1`] = `
<div
  id="CountsCardModule"
>
  <div
    className="card-container"
  >
    <div
      className="card-content"
    >
      <div
        className="card-inner-content"
      >
        <div
          className="card-count-type hasBorder"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="card-count"
          >
            3
          </div>
          <div
            className="card-type"
          >
            <div>
              Deployables
            </div>
          </div>
        </div>
        <div
          className="card-count-type"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="card-count"
          >
            1
          </div>
          <div
            className="card-type"
          >
            <div>
              Deployments
            </div>
          </div>
        </div>
        <div
          className="card-count-type"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="card-count"
          >
            0
          </div>
          <div
            className="card-type"
          >
            <div>
              Failed Deployments
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
